pkg/hubble/parser/threefour/parser.go

```diff
@@ -496,22 +496,9 @@ func isReply(reason uint8) bool {
 
 func decodeIsReply(tn *monitor.TraceNotify, pvn *monitor.PolicyVerdictNotify) *wrapperspb.BoolValue {
 	switch {
-	case tn != nil && monitorAPI.TraceObservationPointHasConnState(tn.ObsPoint):
-		// Unfortunately, not all trace points have the connection
-		// tracking state available. For certain trace point
-		// events, we do not know if it actually was a reply or not.
-		return &wrapperspb.BoolValue{
-			Value: isReply(tn.Reason),
-		}
-	case tn != nil && tn.ObsPoint == monitorAPI.TraceToNetwork && tn.Reason > 0:
-		// FIXME(GH-18460): Even though the BPF programs emitting TraceToNetwork
-		// do have access to connection tracking state, that state is currently
-		// not exposed to userspace by all trace points. Therefore TraceToNetwork
-		// is currently excluded in TraceObservationPointHasConnState.
-		// However, the NodePort return path in handle_ipv4_from_lxc does
-		// populate tn.Reason, and always has with a non-zero value due it
-		// only being used for replies. Therefore, if tn.Reason is non-zero,
-		// we can safely determine if the traced packet was a reply or not.
+	case tn != nil && monitor.TraceReasonIsKnown(tn.Reason):
+		// Reason was not specified, likely because datapath did not
+		// have the information when emitting the trace.
```

Comment:
nit: I believe this comment here is a bit confusing. In this case, the trace reason should be known I think?

Reply:
I think I was the one confused when writing this comment :thinking: It seems wrong, I'll change it. Thanks!